File: src/main/java/io/github/Earth1283/fixLag/listeners/SpawnerOptimizer.java
package io.github.Earth1283.fixLag.listeners;

import io.github.Earth1283.fixLag.FixLag;
import io.github.Earth1283.fixLag.managers.ConfigManager;
import org.bukkit.Bukkit;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.SpawnerSpawnEvent;

public class SpawnerOptimizer implements Listener {

    private final FixLag plugin;
    private final ConfigManager config;

    public SpawnerOptimizer(FixLag plugin) {
        this.plugin = plugin;
        this.config = plugin.getConfigManager();
    }

    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
    public void onSpawnerSpawn(SpawnerSpawnEvent event) {
        if (!config.isSpawnerOptimizerEnabled()) return;

        double tps = Bukkit.getServer().getTPS()[0];
        if (tps < config.getSpawnerOptimizerTpsThreshold()) {
            event.setCancelled(true);
        }
    }
}

File: src/main/kotlin/io/github/Earth1283/fixLag/listeners/SpawnerOptimizer.kt
package io.github.Earth1283.fixLag.listeners

import io.github.Earth1283.fixLag.FixLag
import org.bukkit.Bukkit
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.entity.SpawnerSpawnEvent

class SpawnerOptimizer(plugin: FixLag) : Listener {

    private val config = plugin.configManager

    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
    fun onSpawnerSpawn(event: SpawnerSpawnEvent) {
        if (!config.isSpawnerOptimizerEnabled) return

        val tps = Bukkit.getServer().tps[0]
        if (tps < config.spawnerOptimizerTpsThreshold) {
            event.isCancelled = true
        }
    }
}
